windows-containers/windows-monitoring.md

```diff
@@ -0,0 +1,165 @@
+  `openshift.io/cluster-monitoring=true` label so that cluster monitoring
+  stack will pick up the service monitor in the next steps.
+* Create service, endpoint in `openshift-windows` pointing to WMI exporter 
+  endpoint
+* Create a service monitor in `openshift-windows` namespace for service
+  created above
+
+The inputs to this playbook
+* The internal ip address of the Windows Node where we want the WMI exporter
+  to run.
+* URL from which WMI msi can be downloaded from
+
+### Justification
+
+The reason for having ansible playbook instead of operator with a container
+image is that all Windows container images contains a Windows Kernel and 
+Red Hat has a policy not to ship 3rd party kernels for support reasons. Please
+look at [alternatives](#Alternatives) section
+
+### Risks and Mitigations
+
+The main risks with this proposal are as follows:
+* We are depending on Microsoft to provide us supported version of WMI. As of 
+  now, WMI is hosted at a repo managed by 
+  [individual](https://github.com/martinlindhe/wmi_exporter).
+  
+  
+Mitigation
+* If Microsoft doesn't provide us with WMI, we would just go ahead with the 
+  upstream repo and start using it.
+
+## Design Details
+
+### Test Plan
+
+We plan to add e2e tests to ensure
+* Service, endpoints and Service Monitors are properly created
+* Prometheus is able to collect data from WMI exporter  
+
+### Graduation Criteria
+
+This enhacement will start as GA
+
+### Upgrade / Downgrade Strategy
+
+We will support upgrades/downgrades by publishing a new release of Windows Node
+Monitoring playbook. An older release of the playbook can be used to downgrade.
+
+## Implementation History
+
+v1: Initial Proposal
+
+## Drawbacks
+
+Running WMI exporter as a Windows Service instead of running as a DaemonSet pod
+makes it hard for the Prometheus to monitor. The limitation of not able to run
+WMI exporter on Windows nodes as a pod is because of 
+* Lack of previleged containers on Windows nodes 
+* Windows doesn't write the proc, cpu and mem stats to single file
+
+So, in order to overcome the limitation we're running WMI exporter as a Windows
+service but that makes the monitoring of the Windows Nodes harder.
+ 
+
+## Alternatives
+
+An alternative approach would be write an operator which is responsible for 
+installing WMI exporter to Windows nodes and ensures that it runs as a
+Windows Service and distribute this operator as a Windows binary. The downsides
+to this approach are 
+* Once Windows and WMI exporter community reach a stage where WMI can run as a 
+  container, we can leverage the cluster monitoring operator to create a 
+  DaemonSet like node_exporter for linux and we wouldn't need operator to 
+  manage it.
+* The operator needs to get credentials with enough privileges to create 
+  services, endpoint, service monitors
+
+Considering the above downsides, timeframe and all the unknowns present in this
+project, we have decided not to go with operator based model
```

Comment:
I think for something that is supposed to immediately be GA this I'd want this to be explicitly approved by @smarterclayton or @derekwaynecarr. I don't think falling back into old patterns is a good idea, but I also lack some amount of perspective around windows support, so if there is consensus this is a good idea, then I won't stand in its way.

Reply:
cc: @sdodson @crawford @mpbarrett

Reply:
https://github.com/openshift/enhancements/blob/master/enhancements/windows-containers/productization.md#justification details why we cannot ship these components as containers and as a result they cannot be managed via operators.